### ISSUE TITLE
error in nextjs vercel edge runtime

### ISSUE THREAD

**Issue by MiraiTunga** [josdejong/mathjs#3001]
**Describe the bug**
error occurs when an mathsjs function is used in nextjs vercel edge runtime

```
error Error [TypeError]: Argument to 'typed' at index 1 is not a (typed) function, nor an object with signatures as keys and functions as values.
    at <unknown> (webpack-internal:///(sc_server)/./node_modules/typed-function/lib/umd/typed-function.js:1784)
    at typed (webpack-internal:///(sc_server)/./node_modules/typed-function/lib/umd/typed-function.js:1779:31)
    at eval (webpack-internal:///(sc_server)/./node_modules/mathjs/lib/esm/function/arithmetic/unaryPlus.js:42:14)
    at assertAndCreate (webpack-internal:///(sc_server)/./node_modules/mathjs/lib/esm/utils/factory.js:45:16)
    at eval (webpack-internal:///(sc_server)/./node_modules/mathjs/lib/esm/entry/pureFunctionsAny.generated.js:616:99)
```

**To Reproduce**
use mathjs in nextjs vercel edge function

Important any mathsjs module in an nextjs edge function 
You will get the error above

Does mathsjs use any of these unsupported APIs ?https://edge-runtime.vercel.sh/features/available-apis#unsupported-apis

**Comment by josdejong** [comment]
> use mathjs in vercel edge runtime

How?

**Comment by MiraiTunga** [comment]
@josdejong I have updated the issue

**Comment by josdejong** [comment]
Ah, that gives a good clue, thanks. I have no experience with Vercel Edge, so I guess I would have to figure that out, create an account, etc.

The reason is probably that the mathjs bundle contains an occurrence of `new Function("return this")`, which is not allowed by Vercel Edge (it is a CSP issue). This code is not part of mathjs itself, but is introduced by the Webpack when creating the bundle.

Solution is to configure webpack such that it doesn't output the `new Function` in the bundle.

For reference:

- https://github.com/webpack/webpack/discussions/16734
- https://stackoverflow.com/questions/48695579/how-to-remove-eval-and-function-constructor-from-webpack-build-to-avoid-csp-issu

**Comment by josdejong** [comment]
I expect this issue will be fixed via d0d11a2 (not yet published)
